### PR TITLE
doc: update hydration doc, add flowchart

### PR DESCRIPTION
**Pull request by alaincormier-ssc**
- update hydration process documentation
- add flowchart

### CODE REVIEW

File: Architecture/Repository Structure.md [hunk 1]
```diff
@@ -65,16 +65,20 @@ To overcome this limitation and checkout a specific tag or commit SHA, run `modu
...
+The tools submodule contains a `hydrate.sh` script to hydrate the configs with `kpt`.  The script must be executed when any changes to `source-base` and/or `source-customization` are made.  It can be executed locally or in a pre-commit hook and also in a validation pipeline during PR creation.
 
 At a high level, the script will:
 
-1. For each environment (experimentation, dev, preprod, prod), check if `source-customization/` contains that sub directory.  If so:
+1. Find properly stuctured 'deploy', 'source-base' and 'source-customization' directories, then for each environment (experimentation, dev, preprod, prod), check if `source-customization/` contains that sub directory.  If so:
     - Create a `temp-workspace/<env>` directory to copy the `source-base/` and then copy `source-customization/<env>/`, this adds customization specific to that environment.
+    - Validate that setters file are customized.
```
Comment (lucstjean-ssc) [inline Architecture/Repository Structure.md:74]: Maybe `file` should be plural if you meant multiple setters files?

Comment (alaincormier-ssc) [reply Architecture/Repository Structure.md:74]: yeah, could be zero, one or many...
how about `file(s)`?

**Review by lucstjean-ssc — APPROVED** [review]
/lgtm

File: Architecture/Repository Structure.md (hunk 1 above)
Comment (alaincormier-ssc) [reply Architecture/Repository Structure.md]: done!

**Review by ovidiuasiminei-ssc — APPROVED** [review]
/lgtm

**Review by davelanglois-ssc — APPROVED** [review]

**Pull request merged**